apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/SamhsaMatcherFromClaimTransformerTest.java

```diff
@@ -0,0 +1,649 @@
+   * should be a match for this combination.
+   *
+   * @param system the system value of the first coding
+   * @param code the code of the first coding
+   * @param system2 the system value of the second coding
+   * @param code2 the code of the second coding
+   * @param shouldMatch if the matcher should match on this combination
+   * @param explanationOfBenefit the explanation of benefit
+   */
+  private void verifySamhsaMatcherForItemWithMultiCoding(
+      String system,
+      String code,
+      String system2,
+      String code2,
+      boolean shouldMatch,
+      ExplanationOfBenefit explanationOfBenefit) {
+
+    ExplanationOfBenefit modifiedEob = explanationOfBenefit.copy();
+
+    // Set Top level diagnosis and package code to null so we can test item logic
+    for (ExplanationOfBenefit.DiagnosisComponent diagnosisComponent : modifiedEob.getDiagnosis()) {
+      CodeableConcept codeableConcept = diagnosisComponent.getDiagnosisCodeableConcept();
+      codeableConcept.setCoding(new ArrayList<>());
+      diagnosisComponent.setPackageCode(null);
+    }
+
+    List<Coding> codings = new ArrayList<>();
+    Coding coding = new Coding();
+    coding.setSystem(system);
+    coding.setCode(code);
+    Coding coding2 = new Coding();
+    coding2.setSystem(system2);
+    coding2.setCode(code2);
+    codings.add(coding);
+    codings.add(coding2);
+    modifiedEob.getItem().get(0).getService().setCoding(codings);
+
+    assertEquals(shouldMatch, samhsaMatcher.test(modifiedEob));
+  }
+
+  /**
+   * Verify SAMHSA matcher for ICD item with the given system, code and if the expectation is that
+   * there should be a match for this combination.
+   *
+   * @param system the system value
+   * @param code the code
+   * @param shouldMatch if the matcher should match on this combination
+   */
+  private void verifySamhsaMatcherForDiagnosisIcd(
+      String system, String code, boolean shouldMatch, ExplanationOfBenefit explanationOfBenefit) {
+
+    ExplanationOfBenefit modifiedEob = explanationOfBenefit.copy();
+
+    // Set diagnosis
+    for (ExplanationOfBenefit.DiagnosisComponent diagnosisComponent : modifiedEob.getDiagnosis()) {
+      CodeableConcept codeableConcept = diagnosisComponent.getDiagnosisCodeableConcept();
+      if (codeableConcept != null) {
+        ArrayList<Coding> codingList = new ArrayList<>();
+        codingList.add(new Coding().setSystem(system).setCode(code));
+        codeableConcept.setCoding(codingList);
+        diagnosisComponent.setPackageCode(null);
+      }
+    }
+
+    // Set procedure to empty so we dont check it for matches
+    for (ExplanationOfBenefit.ProcedureComponent diagnosisComponent : modifiedEob.getProcedure()) {
+      CodeableConcept codeableConcept = diagnosisComponent.getProcedureCodeableConcept();
+      ArrayList<Coding> codingList = new ArrayList<>();
+      codeableConcept.setCoding(codingList);
+    }
```

Comment:
Small thing, but this seems a bit verbose, I think it could just be
```java
modifiedEob.getProcedure().forEach(c -> c.getProcedureCodeableConcept().setCoding(new ArrayList<>()));

Reply:
Adjusted!